Bump Ark to 0.1.199

lionel- opened the pull request:
Big Ark update related to treatment of document and workspace symbols:

- https://github.com/posit-dev/ark/pull/855
  Closes https://github.com/posit-dev/ark/issues/846
  Addresses https://github.com/posit-dev/positron/issues/6137

- https://github.com/posit-dev/ark/pull/856
  Addresses https://github.com/posit-dev/positron/issues/1428

- https://github.com/posit-dev/ark/pull/858 and https://github.com/posit-dev/ark/pull/860
  Addresses https://github.com/posit-dev/positron/issues/6546
  Addresses https://github.com/posit-dev/positron/issues/6107
  Addresses https://github.com/posit-dev/positron/issues/5241

- https://github.com/posit-dev/ark/pull/859
  Addresses https://github.com/posit-dev/positron/issues/8330

- https://github.com/posit-dev/ark/pull/861
  Addresses https://github.com/posit-dev/positron/issues/6549

- https://github.com/posit-dev/ark/pull/866
  Addresses https://github.com/posit-dev/positron/issues/4886

- https://github.com/posit-dev/ark/pull/867
  Addresses https://github.com/posit-dev/positron/issues/8402

Fix from @jennybc:

- https://github.com/posit-dev/ark/pull/877
  Addresses https://github.com/posit-dev/positron/issues/5486

Other updates:

- https://github.com/posit-dev/ark/pull/865
- https://github.com/posit-dev/ark/pull/873  (@isabelizimm)


### Release Notes

#### New Features

- `test_that()` blocks now appear in the outline and breadcrumbs, and also support sticky scroll functionality (https://github.com/posit-dev/positron/issues/1428).

- All functions passed as named arguments also gain outline, breadcrumbs, and sticky scroll support. This improves the UX for R6 methods in particular (https://github.com/posit-dev/positron/issues/6546).

- Added support for comment sections (like `# Title ----`) in function calls. This is especially useful for users of the `targets` package where it is common to have very large targets calls that can now be sectioned in the outline, with breadcrumbs and sticky scroll support (https://github.com/posit-dev/positron/issues/8402).

- R6 methods are now exported as workspace symbols. You now search for them with `Ctrl/Cmd + t` (https://github.com/posit-dev/positron/issues/6549).

- New `positron.r.symbols.includeAssignmentsInBlocks` setting to control whether local variables in functions and nested `{}` blocks (e.g. in `test_that()` calls) appear in the outline. It's turned off by default to keep the outline focused on the overall structure of the file (https://github.com/posit-dev/positron/issues/8330).

- New `positron.r.workspaceSymbols.includeCommentSections` setting to control whether comment sections (like `# Title ----`) are exported as workspace symbols. When enabled, you can search for the sections with `Ctrl/Cmd + t`. This is disabled by default to keep symbol search focused on actual symbols (https://github.com/posit-dev/positron/issues/4886).

- `browseURL()` in R now delegates to the operating system's default opener for inputs that are not recognized as a web URL or an HTML file (https://github.com/posit-dev/positron/issues/5486).

#### Bug Fixes

- The breadcrumbs and sticky scroll features now behave properly in the presence of comment sections like `# Title ----` (https://github.com/posit-dev/positron/issues/6137).

- Section comments nested in `{ }` blocks now appear in the outline https://github.com/posit-dev/positron/issues/6107.

- When a comment section (like `# Title ----`) is emitted as workspace symbol, and it clashes with another top-level definition (such as `Title <- function() {}`), the "Go to definition" feature now always prioritizes the definition of regular functions and variables (https://github.com/posit-dev/positron/issues/4886).


### QA Notes

See linked Ark PRs.

I had to update E2E tests to:

- Avoid making assumptions about whether local variables are included in the outline. This is now a user setting and we're still discussing the default. Even if we don't turn it off by default (which was the initial assumption), better not have E2E tests depend on that.

- Outline elements no longer retrieve metadata, only the symbol name. The metadata seems too brittle to changes in backends, for instance the variables now include a summary of related problems, and this was included in the element string, causing a test failure.

@:ark
@:outline

lionel- committed:
Bump Ark to 0.1.198

lionel- committed:
Update outline test

lionel- committed:
Bump to Ark 0.1.199

lionel- committed:
Only extract symbol name from the outline

testlabauto reviewed (dismissed):
Thank you for updating the tests!

lionel- committed:
Add `positron.r.symbols.includeAssignmentsInBlocks` setting (#8412)

For https://github.com/posit-dev/positron/issues/8330
Extension side of https://github.com/posit-dev/ark/pull/859

Can be merged separately.
